doc/requirements/requirements.md

```diff
@@ -0,0 +1,306 @@
+
+![Command Pipeline with Support](./images/Command-Pipeline-Support.png)
+
+To facilitate error recovery, each module in the chain, if it is unable to carry out its task, its able to propagate error information to its predecessor in the command chain.
+
+## 1.5 Design Goals
+
+The Navigation System designers should strive to meet the following high-level design goals:
+
+* **Modularity** - The Navigation Stack should allow for other implementers to easily replace architectural level components with alternative implementations, in particular, the global and local planners.
+* **Portability** - The Navigation Stack should not introduce inherent limitations in the architectural blocks. For example, it should support multiple kinds of robots, not making assumptions about robot capabilities and limitations and should support various map types and orientations.
+* **Performance** - *TODO: What are the performance goals?* 
+* **Scalability** - *TODO: How low should the implementation scale?* *Specify a minimum platform?*
+* *Other important design goals?*
+
+In particular, there are several specific goals for the ROS 2 Navigation System with respect to improving the existing ROS navigation stack:
+
+* The global and local planners should be easy to replace and chain together in a dynamic way. For example, a user may want to have a global planner for classic point-A-to-point-B travel, but upon reaching point B, have a special planner then control a series of maneuvers such as docking to a charging station or a conveyor belt. 
+* The user should be able to plan complex missions which include a series of movements, and the Navigation System should be able to accept those mission plans and execute them. 
+* The user should be able to specify different types of Robot drive types, such as Ackerman (automobile) steering, and Robot shapes (the current navigation stack is very limited in these areas).
+* The Navigation System should be able to handle more map types and orientations, including outdoor and 2D+ terrain.
+
+# 2.0 Requirements
+
+This section lists the requirements for the Navigation System.
+
+## 2.1 Implementation Constraints
+
+There are various constraints on the development of the ROS 2 Navigation stack.
+
+Id | Handle | Priority | Description | Notes
+-- | ------ | -------- | ----------- | -----
+IC001 | Developer's Guide | 1 | The Navigation System SHALL be developed in accordance with the ROS 2 Devloper's Guide | [ROS 2 Developer's Guide](https://github.com/ros2/ros2/wiki/Developer-Guide)
+IC002 | Implementation Language.C++.Version | 1 | Developers SHALL assume the availability of C++14 language features | Per the ROS 2 Developer's Guide
+IC003 | Implementation Language.C++.API Preference | 1 | Developers SHALL prefer standard C++, then Boost, then custom code, in that order. | Boost may be used if equivalent functionality is not already available in the C++ standard library
```

Comment:
Another SHOULD. I can imagine circumstances where the Boost or Standard Library options are less convenient than custom code, or where the custom code is so small it doesn't make sense to bring in a additional boost dependency.

Reply:
Does it make sense to have this section at all or just point to the ROS2 standard and only call out any additions or exceptions to that standard that we're making? Aren't IC002-IC006 already covered in the ROS2 standard? Sorry I haven't looked specifically for these.

Reply:
Yeah, we could take that approach. I was calling out a few key items for emphasis.